engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManager.scala

```diff
@@ -189,31 +205,48 @@ object K8sDeploymentManager {
 
   val scenarioVersionAnnotation: String = "nussknacker.io/scenarioVersion"
 
+  val configMapIdLabel: String = "nussknacker.io/configMapId"
+
   def apply(modelData: ModelData, config: Config)(implicit ec: ExecutionContext, actorSystem: ActorSystem): K8sDeploymentManager = {
     new K8sDeploymentManager(modelData, config.rootAs[K8sDeploymentManagerConfig])
   }
 
+  /*
+    We use name label to find deployment to cancel/findStatus, we append short hash to reduce collision risk
+    if sanitized names are similar (e.g. they differ only in truncated part or non-alphanumeric characters
+    TODO: Maybe it would be better to just pass ProcessId in findJobStatus/cancel?
+   */
+  private def scenarioNameLabelValue(processName: ProcessName) = {
+    val name = processName.value
+    sanitizeLabel(name, s"-${shortHash(name)}")
+  }
+
+  private[manager] def requirementForName(processName: ProcessName): Requirement = scenarioNameLabel is scenarioNameLabelValue(processName)
+
   /*
     Labels contain scenario name, scenario id and version.
-    We use name label to find deployment to cancel/findStatus, it *won't* work properly if sanitized names
-    are not unique
-    TODO: we need some hash to avoid name clashes :/ Or pass ProcessId in findJobStatus/cancel
    */
-  private[manager] def labelSelectorForName(processName: ProcessName) =
-    LabelSelector(scenarioNameLabel is sanitizeLabel(processName.value))
+  private[manager] def labelsForScenario(processVersion: ProcessVersion) = Map(
+    scenarioNameLabel -> scenarioNameLabelValue(processVersion.processName),
+    scenarioIdLabel -> processVersion.processId.value.toString,
+    scenarioVersionLabel -> processVersion.versionId.value.toString
+  )
 
   /*
     Id of both is created with scenario id + sanitized name. This is to:
     - guarantee uniqueness - id is sufficient for that, sanitized name - not necessarily, as replacement/shortening may lead to duplicates
       (other way to mitigate this would be to generate some hash, but it's a bit more complex...)
     - ensure some level of readability - only id would be hard to match name to scenario
    */
-  private[manager] def objectNameForScenario(processVersion: ProcessVersion): String = {
-    sanitizeObjectName(s"scenario-${processVersion.processId.value}-${processVersion.processName}")
+  private[manager] def objectNameForScenario(processVersion: ProcessVersion, append: String = ""): String = {
```

Comment:
Would be nice to have some unit test checking if for long process name with the same prefix but other hash is generated different object name. I know that you have done it correctly but it is easy to be broken and unit test looks like simple thing to add

Reply:
:+1: added some tests